daemon/logger/syslog/syslog.go

```diff
@@ -55,3 +69,25 @@ func (s *Syslog) Name() string {
 func (s *Syslog) GetReader() (io.Reader, error) {
 	return nil, logger.ReadLogsNotSupported
 }
+
+func parseAddress(address string) (string, string, error) {
+	if urlutil.IsTransportURL(address) {
+		url, err := url.Parse(address)
+		if err != nil {
+			return "", "", err
+		}
+
+		proto := url.Scheme
+		address := url.Host
+		if _, _, err := net.SplitHostPort(address); err != nil {
+			if !strings.Contains(err.Error(), "missing port in address") {
```

Comment:
Sad that the only way to check this error is to test the string...I looked :(

Maybe instead we can check for error, if `err != nil` then append the default port and run `net.SplitHostPort` again, that way we avoid this string check on the error.